### PR TITLE
Update pyO3 to 0.18

### PR DESCRIPTION
**Pull request by g-bauer**
- [x] feos-core
- [x] feos-dft
- [x] feos-derive
- [x] feos

### CODE REVIEW

**Review by prehner — COMMENTED** [review]
Excellent! Some things are much clearer and also more easily maintainable for the future.

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - The limitations of the homo gc method for PC-SAFT are enforced more strictly. [#88](https://github.com/feos-org/feos/pull/88)
 - Removed generics for units in all structs and traits in favor of static SI units. [#115](https://github.com/feos-org/feos/pull/115)
 
+### Packaging
+- Updated `pyo3` and `numpy` dependencies to 0.18. [#119](https://github.com/feos-org/feos/pull/119)
```
[the remark this answers is not in the document]
Comment (g-bauer) [reply CHANGELOG.md:22]: Ah correct, thanks!

File: Cargo.toml [hunk 2]
```diff
@@ -22,12 +22,12 @@ members = ["feos-core", "feos-dft", "feos-derive"]
 crate-type = ["rlib", "cdylib"]
 
 [dependencies]
-quantity = "0.5"
-num-dual = "0.5"
+quantity = "0.6"
+num-dual = { version = "0.6", features = ["linalg"] }
```
[the remark this answers is not in the document]
Comment (g-bauer) [reply Cargo.toml:26]: Correct - copy and paste error. Removed the feature.

File: src/association/python.rs [hunk 3]
```diff
@@ -11,6 +11,7 @@ pub struct PyAssociationRecord(pub AssociationRecord);
 
 #[pymethods]
 impl PyAssociationRecord {
+    #[pyo3(signature = (kappa_ab, epsilon_k_ab, na=None, nb=None))]
```
[the remark this answers is not in the document]
Comment (g-bauer) [reply src/association/python.rs:14]: `__text_signature__` for `#[new]` can only be placed at a struct.

File: src/gc_pcsaft/python/mod.rs [hunk 4]
```diff
@@ -106,7 +106,7 @@ impl PyGcPcSaftEosParameters {
 #[cfg(feature = "dft")]
 #[pyclass(name = "GcPcSaftFunctionalParameters")]
 #[pyo3(
-    text_signature = "(pure_records, segmentbinary_records=None, substances=None, search_option='Name')"
+    text_signature = "(pure_records, segmentbinary_records=None, substances=None, search_option=IdentifierOption.Name)"
```
[the remark this answers is not in the document]
Comment (g-bauer) [reply src/gc_pcsaft/python/mod.rs:109]: `signature` can only be placed on functions. Those are defined in the `impl_parameter_from_segments` macro.

**Review by prehner — APPROVED** [review]

**Pull request merged**